rfcs/rfc-006-budget-contract-language.md

```diff
@@ -0,0 +1,117 @@
+
+### Transfer (aka, Budget v2) Transaction Format
+```rust
+pub enum TransferCondition {
+    /// Wait for a `Timestamp` `Instruction` at or after the given `DateTime` from `Pubkey`
+    Timestamp(DateTime<Utc>, Pubkey),
+
+    /// Wait for a `Witness` `Instruction` from `Pubkey`
+    Witness(Pubkey),
+}
+
+pub struct Transfer {
```

Comment:
This is called `Payment` in payment_plan.rs, but includes a `to` pubkey. In the current code, a Budget expression wraps and guards the `Payment` instead of including those guards in the struct. In this proposed form, you'd need to add `cancellable` and `conditions` fields to every type of action, not just payment. Also, if a transaction was not cancellable and had no conditions, you'd have to pay 9 bytes to tell the system its the base case instead of just 1, `Budget::Pay`.